### PR TITLE
(1509) Redirect old domain to new one

### PR DESCRIPTION
**Pull request by pezholio**
This redirects the old domains (https://beis-roda-[staging|prod].london.cloudapps.digital/) to the new domains. Currently, trying to access the site via the old domains results in an error. This adds the old domains to the list of allowed domains, and then redirects them to the canonincal domain. The env vars are ready to go and set in Github, so no extra work is required to deploy this

### CODE REVIEW

File: config/routes.rb [hunk 1]
```diff
@@ -3,10 +3,9 @@
 Rails.application.routes.draw do
   # If the DOMAIN env var is present, and the request doesn't come from that
   # hostname, redirect us to the canonical hostname with the path and query string present
-  if ENV["DOMAIN"].present?
-    hostname = URI(ENV["DOMAIN"]).hostname
-    constraints(host: Regexp.new("^(?!#{Regexp.escape(hostname)})")) do
-      match "/(*path)" => redirect(host: hostname), :via => [:all]
+  if ENV["CANONICAL_HOSTNAME"].present?
+    constraints(host: Regexp.new("^(?!#{Regexp.escape(ENV["CANONICAL_HOSTNAME"])})")) do
```
Comment (mec) [inline config/routes.rb:7]: Cripes! I think I understand this!

If the hostname is not the canonical one (www.report-official-development-assistance.service.gov.uk) we redirect to it for all paths and transports – is that right?

**Review by mec — COMMENTED** [review]
Having the regex redirect seems like the answer to this issue, but it makes me nervous! Probably because I don't understand enough though, so I am having to work extra hard to get confidence here.

I find all these ENV VAR names hard to parse as well, I wonder if other folks do – it could just be me?

I went to look at all at the ENV VAR and tfvar files in 1Password as the context helps me parse all this - but it doesn't look they have been updated - I think it's super important we do that and I am sure you were planning to once merged – maybe talking about them in the commit messages would have helped me?

I would like to verify I have understood all this correctly and be sure we've done all we can to make these VARs as clear and as simple as we can, happy to chat about that here or elsewhere?

File: config/routes.rb (hunk 1 above)
Comment (pezholio) [reply config/routes.rb:7]: Yep, that's right 👍

Comment (jcoglan) [reply config/routes.rb:7]: This feels like logic that I'm more used to putting in nginx config, or a similar "infrastructure" location, rather than in the app -- is there somewhere more appropriate in our infrastructure for dealing with host-based redirects?

Comment (pezholio) [reply config/routes.rb:7]: I don't think we have that luxury, given we're containerised.

File: terraform/app.tf [hunk 2]
```diff
@@ -18,8 +18,9 @@ resource "cloudfoundry_app" "beis-roda-app" {
     "RAILS_ENV"                              = "production"
     "DATA_MIGRATE"                           = var.data_migrate
     "SECRET_KEY_BASE"                        = var.secret_key_base
-    "DOMAIN"                                 = var.domain
-    "ADDITIONAL_DOMAINS"                     = var.additional_domains
+    "CANONICAL_HOSTNAME"                     = var.canonical_hostname
+    "DOMAIN"                                 = "https://${var.canonical_hostname}"
+    "ADDITIONAL_HOSTNAMES"                   = var.additional_hostnames
```
[the remark this answers is not in the document]
Comment (pezholio) [reply terraform/app.tf:23]: Yeah, maybe `DOMAIN` is not the right naming here. Something like `ROOT_URI` would be more sensible I think.

File: terraform/app.tf [hunk 3]
```diff
@@ -18,8 +18,9 @@ resource "cloudfoundry_app" "beis-roda-app" {
     "RAILS_ENV"                              = "production"
     "DATA_MIGRATE"                           = var.data_migrate
     "SECRET_KEY_BASE"                        = var.secret_key_base
-    "DOMAIN"                                 = var.domain
-    "ADDITIONAL_DOMAINS"                     = var.additional_domains
+    "CANONICAL_HOSTNAME"                     = var.canonical_hostname
+    "DOMAIN"                                 = "https://${var.canonical_hostname}"
```
[the remark this answers is not in the document]
Comment (pezholio) [reply terraform/app.tf:22]: What are those set to at the moment?

[the remark this answers is not in the document]
Comment (pezholio) [reply terraform/app.tf:22]: I feel like `custom_hostname` should be our canonical hostname too. But I'm not sure if it's set the the right value at the mo, becuase it's stored as a secret.

[the remark this answers is not in the document]
Comment (pezholio) [reply terraform/app.tf:22]: This kinda sums up why I expressly didn't make the new env vars a secret in the actions setup. If something doesn't _need_ to be a secret, it shouldn't be, because we end up with this confusion.

File: terraform/app.tf (hunk 2 above)
[the remark this answers is not in the document]
Comment (jcoglan) [reply terraform/app.tf:23]: `DOMAIN` is a more specific term than `URI` -- for example, this cannot be any URL, it can only be a scheme and host. For example it could contain `https://example.com` but not `https://example.com/page.html`. Appropriate terms for this are "domain" or "origin".

If this variable can actually contain arbitrary URLs, then calling it a URL or URI would be fine.

[the remark this answers is not in the document]
Comment (jcoglan) [reply terraform/app.tf:23]: ("domain" is a badly defined word but enough people use it to mean either "hostname" or "origin" or "site" (in the browser security model sense) that it's probably fine.)

[the remark this answers is not in the document]
Comment (mec) [reply terraform/app.tf:23]: No problem - I am happy!

**Review by mec — APPROVED** [review]
Thanks for bearing with me, this looks good! Go DOMAIN! :)
